apps/vm-hotreload/webpack-custom-hmr/src/custom-hmr-helpers.js

```diff
@@ -0,0 +1,858 @@
+const vm = require('vm');
+
+// Webpack runtime code for in-memory HMR - only executed when webpack is available
+function injectInMemoryHMRRuntime(__webpack_require__) {
```

Comment:
copy over the runtime module from HMR plugin - the plugin (in node targets) is designed to read filesystem only to check for updates. I replace some of the methods to allow in-memory patches, so that i could fetch() the update from a KV store or database etc and not need to manually write to disk.